### PR TITLE
feat: automatically change status on using auto increment button

### PR DESCRIPTION
**Pull request by BitForger**
Why?

If I'm in the planning list and I swipe to increment it, I should expect it to change the watching status to current

### CODE REVIEW

File: AniHyou/Screens/MediaList/ViewModels/MediaListViewModel.swift [hunk 1]
```diff
@@ -87,9 +87,9 @@ class MediaListViewModel: ObservableObject {
     
     @Published var isLoading = false
     
-    func updateEntryProgress(entryId: Int, progress: Int) {
+    func updateEntryProgress(entryId: Int, progress: Int, status: GraphQLEnum<MediaListStatus>) {
```
Comment (axiel7) [inline AniHyou/Screens/MediaList/ViewModels/MediaListViewModel.swift:90]: Can you make the status param nullable and change the type to MediaListStatus? So it's not sent if we don't really want to update it. Take a look at the [updateEntry](https://github.com/axiel7/AniHyou/blob/9e4256a949406558817dbeca8a695a99755ee9a3/AniHyou/Screens/MediaDetails/Edit/ViewModels/MediaListEditViewModel.swift#L63) function

File: AniHyou/Screens/MediaList/Views/MediaListView.swift [hunk 2]
```diff
@@ -122,6 +69,79 @@ struct MediaListView: View {
...
+            .tint(.blue)
+        }
+    }
+    
+    func updateEntryProgress(item: UserMediaListQuery.Data.Page.MediaList!) {
+        var status: GraphQLEnum<MediaListStatus>
+        if item.status == .planning {
+            status = .case(.current)
+        } else {
+            status = item.status!
```
Comment (axiel7) [inline AniHyou/Screens/MediaList/Views/MediaListView.swift:132]: here it would be nil

File: AniHyou/Screens/MediaList/Views/MediaListView.swift [hunk 3]
```diff
@@ -122,6 +69,79 @@ struct MediaListView: View {
...
+        }
+        viewModel.updateEntryProgress(entryId: item.id, progress: item.progress! + 1, status: status)
+    }
+    
+    func shouldShowIncrementButton() -> Bool {
```
Comment (axiel7) [inline AniHyou/Screens/MediaList/Views/MediaListView.swift:137]: I think a calculated var would be better here, but you can leave as it

File: AniListAPI/Sources/LocalCacheMutations/BasicMediaListEntry.graphql.swift [hunk 4]
```diff
@@ -113,7 +113,7 @@ public struct BasicMediaListEntry: AniListAPI.MutableSelectionSet, Fragment {
         "completedAt": completedAt._fieldData,
       ],
       fulfilledFragments: [
-        ObjectIdentifier(BasicMediaListEntry.self)
+        ObjectIdentifier(Self.self)
```
Comment (axiel7) [inline AniListAPI/Sources/LocalCacheMutations/BasicMediaListEntry.graphql.swift:116]: what's your apollo version? I think you should update the CLI by installing it again. This [changes](https://github.com/axiel7/AniHyou/commit/f2e0c86b57a24438bfdb3d83c51890a9b9bbd8f5) were caused by updating Apollo to 1.2.2

File: AniHyouShou Watch App/MediaList/UpdateMediaEntryView.swift [hunk 5]
```diff
@@ -24,7 +24,7 @@ struct UpdateMediaEntryView: View {
                     Text("\(entry!.progress ?? 0)/\(entry!.totalProgress ?? 0)")
 
                     Button(action: {
-                        viewModel.updateEntryProgress(entryId: entry!.id, progress: (entry!.progress ?? 0) + 1)
+                        viewModel.updateEntryProgress(entryId: entry!.id, progress: (entry!.progress ?? 0) + 1, status: entry!.status ?? .case(.current))
```
Comment (axiel7) [inline AniHyouShou Watch App/MediaList/UpdateMediaEntryView.swift:27]: here it would be also nil

**Review by axiel7 — CHANGES_REQUESTED** [review]

File: AniListAPI/Sources/LocalCacheMutations/BasicMediaListEntry.graphql.swift (hunk 4 above)
Comment (BitForger) [reply AniListAPI/Sources/LocalCacheMutations/BasicMediaListEntry.graphql.swift:116]: xcode 😬  yeah I think it decided 1.2.1 was the one it should use. i'll fix it

File: AniHyou/Screens/MediaList/Views/MediaListView.swift (hunk 3 above)
Comment (BitForger) [reply AniHyou/Screens/MediaList/Views/MediaListView.swift:137]: ahh you're right... javascript roots are showing 😩

File: AniHyou/Screens/MediaList/ViewModels/MediaListViewModel.swift (hunk 1 above)
Comment (BitForger) [reply AniHyou/Screens/MediaList/ViewModels/MediaListViewModel.swift:90]: yep. I was trying to figure out how to do it. I'll give it another crack

**Review by axiel7 — APPROVED** [review]

**Pull request merged**